docs/jemalloc_usage.md

```diff
@@ -0,0 +1,29 @@
+# JEMalloc usage
+FEX-Emu uses two different jemalloc heap allocators at once.
+- jemalloc: The primary heap allocator
+- jemalloc_glibc: The second heap allocator
+
+## jemalloc - primary heap allocator
+This heap allocator is the primary allocator that FEX-Emu uses. All of FEXCore's `fextl::` namespaced objects will allocate memory through this
+allocator. This allocator has its usage of mmap and munmap overridden to use FEXCore's internal VMA region allocator.
+
+### FEXCore internal VMA region allocator
+- When running a 64-bit guest application, this VMA region allocator is disabled and passes through to mmap and munmap in the host kernel.
+- This only gets interesting when running a 32-bit guest application.
+
+When running a 32-bit guest application, the VMA region allocator will allocate all memory above 4GB virtual address space (Up to 256TB on 48-bit VA
+systems!). It then provides a mmap and munmap interface to allow allocating memory out of this reserved virtual address space. This ensures that all
+of FEX's allocations stay out of the lower 32-bit 4GB VA space, otherwise games would quickly run out of virtual address space.
+
+## jemalloc_glibc - secondary heap allocator
+This heap allocator replaces glibc's allocator using weak symbol overriding. This explicitly does **not** use the FEXCore VMA region allocator. This
```

Comment:
> This explicitly does **not** use the FEXCore VMA region allocator

This is backwards again: It's starting to explain something by explaining what it's not.